src/Native/Runtime/ThunksMapping.cpp

```diff
@@ -209,7 +209,68 @@ EXTERN_C REDHAWK_API void* __cdecl RhAllocateThunksMapping()
     return pThunksSection;
 }
 
-#else // FEATURE_RX_THUNKS
+// FEATURE_RX_THUNKS
+#elif FEATURE_FIXED_POOL_THUNKS
+
+// This thread local variable is used for delegate marshalling
+DECLSPEC_THREAD intptr_t tls_thunkData;
+
+extern "C" uintptr_t g_pThunkData;
+uintptr_t g_pThunkData = NULL;
+
+#include "gcenv.os.h"
+
+COOP_PINVOKE_HELPER(int, RhpGetThunkBlockCount, ());
+COOP_PINVOKE_HELPER(int, RhpGetNumThunkBlocksPerMapping, ());
+COOP_PINVOKE_HELPER(int, RhpGetThunkBlockSize, ());
+COOP_PINVOKE_HELPER(void*, RhpGetThunkDataBlockAddress, (void* addr));
+COOP_PINVOKE_HELPER(void*, RhpGetThunkStubsBlockAddress, (void* addr));
+
+EXTERN_C REDHAWK_API void* __cdecl RhAllocateThunksMapping()
+{
+    static int nextThunkDataMapping = 0;
+
+    int thunkBlocksPerMapping = RhpGetNumThunkBlocksPerMapping();
+    int thunkBlockSize = RhpGetThunkBlockSize();
+    int blockCount = RhpGetThunkBlockCount();
+
+    ASSERT(blockCount % thunkBlocksPerMapping == 0)
+
+    int thunkDataMappingSize = thunkBlocksPerMapping * thunkBlockSize;
+    int thunkDataMappingCount = blockCount / thunkBlocksPerMapping;
+
+    if (nextThunkDataMapping == thunkDataMappingCount)
+    {
+        return NULL;
+    }
+
+    if (g_pThunkData == NULL)
+    {
+        int thunkDataSize = thunkDataMappingSize * thunkDataMappingCount;
+
+        g_pThunkData = (uintptr_t)GCToOSInterface::VirtualReserve(thunkDataSize, 4096, VirtualReserveFlags::None);
```

Comment:
`GCToOSInterface` is not meant to be used outside GC. Can this use PalVirtualAlloc? Or is there a particular reason why you choose to use GCToOSInterface here?

Reply:
Just didn't though about PalVirtualAlloc here. The GCToOSInterface interface with separated functions for reserve and commit was just convenient. I should be able to change it to PalVirtualAlloc